src/ucs/sys/sock.h

```diff
@@ -542,20 +542,16 @@ int ucs_sockaddr_ip_cmp(const struct sockaddr *sa1, const struct sockaddr *sa2);
 /**
  * Checks if two IP addresses are in the same subnet
  *
- * @param [in]  sa1         Pointer to sockaddr structure #1.
- * @param [in]  sa2         Pointer to sockaddr structure #2.
+ * @param [in]  sa1         Pointer to sockaddr_storage structure #1.
+ * @param [in]  sa2         Pointer to sockaddr_storage structure #2.
  * @param [in]  prefix_len  Subnet prefix length in bits.
- * @param [out] is_same_p   Pointer to a boolean value indicating if both
- *                          addresses are in the same subnet.
  *
- * @return UCS_OK on success or one of the following errors:
- *         UCS_ERR_INVALID_PARAM - wrong or mismatched address families.
- *         UCS_ERR_EXCEEDS_LIMIT - prefix_len is too big (exceeds size of
- *                                 address).
+ * @return 1 if both addresses are in the same subnet.
+ *         0 if not
  */
-ucs_status_t ucs_sockaddr_is_same_subnet(const struct sockaddr *sa1,
-                                         const struct sockaddr *sa2,
-                                         unsigned prefix_len, int *is_same_p);
+int ucs_sockaddr_is_same_subnet(const struct sockaddr_storage *sa1,
+                                const struct sockaddr_storage *sa2,
```

Comment:
IMO better keep it as const struct sockaddr * since it's input parameter and not output